src/ext/rpmmd.rs

```diff
@@ -0,0 +1,60 @@
+use ::repo::Repo;
+use libc;
+use std::ptr;
+use std::os::unix::io::*;
+use std::fs::File;
+
+pub trait RpmMd {
+
+    fn add_repomd(&mut self, file: & File) -> bool;
+
+    fn add_repomdxml(&mut self, file: & File) -> bool;
+}
+
+impl RpmMd for Repo {
+    fn add_repomd(&mut self, file: &File) -> bool {
```

Comment:
That is the plan, but right now I'm trying to get all the way through depchase before I start refactoring